dev/src/document-change.ts

```diff
@@ -36,7 +41,9 @@ export class DocumentChange {
    * @param {number} newIndex - The index in the documents array after this
    * change.
    */
-  constructor(type, document, oldIndex, newIndex) {
+  constructor(
+      type: string, document: QueryDocumentSnapshot, oldIndex: number,
```

Comment:
Can you just make these: `private readonly _type:string, private readonly _document ...`?

And we you want to be really pedantic, we could change the type of `type` to:
`DocumentChangeType = 'added' | 'removed' | 'modified'`